### PR TITLE
refactoring

### PR DESCRIPTION
**Pull request by philipcraig**
* make file-system location unreachable
* iterate idiomatically
* make pattern match exhaustive (no `_`)

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -497,7 +497,7 @@ mod tests {
                 Statement::Return(_) => {
                     // TODO: Check some property of the expression.
                 }
-                _ => panic!("Expected a return statement."),
```
Comment (jbcoe) [inline src/parser.rs]: Q: What's the motivation for this change? I'm not sure how it's better as adding new statements will mean this needs updating needlessly. I'm sure I'm missing something.

File: src/parser.rs [hunk 2]
```diff
@@ -543,7 +543,7 @@ mod tests {
                     assert_eq!(let_statement.mutable, test_case.mutable);
                     // TODO: Check some property of the expression.
                 }
-                _ => panic!("Expected a let statement"),
+                Statement::Return(_) => panic!("Expected a let statement"),
```
Comment (jbcoe) [inline src/parser.rs:599]: Q: What's the motivation for this change? I'm not sure how it's better as adding new statements will mean this needs updating needlessly. I'm sure I'm missing something.

File: src/parser.rs (hunk 1 above)
Comment (philipcraig) [reply src/parser.rs]: Making the pattern match explicit and exhaustive means code takes into account all possible values. Once there are more than two kinds of statement, clippy will stop complaining I think

File: src/parser.rs (hunk 2 above)
Comment (philipcraig) [reply src/parser.rs:599]: Making the pattern match explicit and exhaustive means code takes into account all possible values. Once there are more than two kinds of statement, clippy will stop complaining I think

**Review by jbcoe — APPROVED** [review]
